core/domain/exp_services.py

```diff
@@ -1710,6 +1710,475 @@ def is_voiceover_change_list(change_list):
+                                    changed_properties[old_state_name]):
+                                change_is_mergeable = True
+                    else:
+                        change_is_mergeable = True
+                # Customization args differ for every interaction, so in
+                # case of different interactions merging is simply not
+                # possible, but in case of same interaction, the values in
+                # the customization_args are often lists so if someone
+                # changes even one item of that list then determining which
+                # item is changed is not feasible, so suppose there is long
+                # list of values in item selection interaction and one user
+                # deletes one value and another one edits another value,
+                # so after deletion the indices of all the values will be
+                # changed and it will not be possible to compare and know
+                # that which value is changed by second user.
+                # So we will not be handling the merge on the basis of
+                # individual fields.
+                elif (change.property_name ==
+                      exp_domain.STATE_PROPERTY_INTERACTION_CUST_ARGS):
+                    if old_state_name in changed_properties:
+                        if (frontend_version_exploration.states[old_state_name].interaction.id == # pylint: disable=line-too-long
+                                backend_version_exploration.states[new_state_name].interaction.id): # pylint: disable=line-too-long
+                            if (all(property not in
+                                    changed_properties[old_state_name]
+                                    for property in ['solution',
+                                                     'recorded_voiceovers',
+                                                     'answer_groups',
+                                                     'widget_customization_args'
+                                                    ]) and
+                                    change.property_name not in
+                                    changed_translations[old_state_name]):
+                                change_is_mergeable = True
+                    else:
+                        change_is_mergeable = True
+                elif (change.property_name ==
+                      exp_domain.STATE_PROPERTY_INTERACTION_ANSWER_GROUPS):
+                    if old_state_name in changed_properties:
+                        if (frontend_version_exploration.states[old_state_name].interaction.id ==  # pylint: disable=line-too-long
+                                backend_version_exploration.states[new_state_name].interaction.id): # pylint: disable=line-too-long
+                            if (all(property not in
+                                    changed_properties[old_state_name]
+                                    for property in ['solution',
```

Comment:
ditto: use named constant

Reply:
here and below.

Reply:
Add clear comments indicating what the lists represent and when/how they should be modified.

Reply:
Made the changes.